crates/bevy_math/src/interpolation/utils.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::{Add, Div, Mul, Sub};
+
+// TODO: evaluate the more accurate algorithm of (Walcyzk et al. 2018.)[https://arxiv.org/abs/1802.06302]
+/// Quake 3 fast inverse sqrt
+///
+/// Implementation borrowed from Piston under the MIT License: [https://github.com/PistonDevelopers/skeletal_animation]
+#[inline]
+pub fn fast_inv_sqrt(x: f32) -> f32 {
```

Comment:
Alright, I did some more digging: Rust's native solution for this is faster than the approximate solution because inverse square root has a dedicated instruction these days. See: https://stackoverflow.com/a/59082210/3234149 

IMO this function should be cut completely and its usages should be replaced with `sqrt().recip()`.

Reply:
Good catch

Reply:
Do you know how the intrinsic rsqrt + a single Newton-Raphson iteration compares with the 'carmack' method?

In that same thread I found this: https://stackoverflow.com/questions/31555260, so I compiled a few possible methods here https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=ddc10be842400c95b81a0c5fba218da6;

Maybe my memory is failing me but a old bench showed me that `sqrtss` as a very hot instruction and that made me use  the carmak method, I will have to double check that ...

Reply:
Small note: While `sqrt` and `1.0/x` are guaranteed under IEEE-754 2008, implementations at the hardware level may or may not be compliant. My particular game I'm working on requires animation to be in the networking simulation hotpath, and cross-platform determinism is mandatory. If we are going down the route of `sqrt().recip()`, could we use `libm::sqrtf` here under a feature flag? glam already has a feature flag for it, so we could pass the feature flag through for ease of use.

The 'carmack' method here does not have this issue and should be consistent across platforms.

Reply:
Yeah, I think we're going to need to do our own modern benchmarks here to get a firm answer on what's correct for this context.

The note on determinism is important; the approach James suggests seems very reasonable.

Reply:
My benches gave-me these results:

| function | time | maximum relative error |
| ---------| ----- | ----------------------- |
| fpu (sqrt + div) | 1.4637 ns | 0% |
| quake3 | 1.1379 ns | 0.17507957% |
| _mm_rsqrt_ss | 846.43 ps | 0.029162358% |
| _mm_rsqrt_ss + 1nr iteration | 1.0436 ns | 0.000020642221%  |

Reply:
Awesome; those are good numbers to have. How hard would it be to ensure input alignment?

I'm in favor of more accuracy + smaller, more legible code base over the micro-optimization here, but I'll defer to others' judgement in the final review.

Reply:
Directly using `_mm_rsqrt_ss` may affect portability, we'll need to find alternatives for non-x86 platforms (i.e. Android/iOS). We'll probably need some SIMD abstraction for this like `simba`. Easiest way to ensure alignment in Rust is to use a `#[repr(C, align(16)]` types like glam's Vec3A/Vec4. 

Given these results, I'm actually more in favor for just using the fast_inv_sqrt as is since it's guaranteed to be bit-for-bit identical across targets, it's a notable speedup over the normal `sqrt().recip()`, and the error is still within tolerable ranges for a lot of the target usecases for such a function.

Reply:
I think in doing `_mm_rsqrt_ss ` with a fallback to `quake3` method for non `x86` or `libm` / `no_std` but I'm not sure if the aligment requiriment will be a problem, expecially when called by the consumer code, on the nlerp side I can make things rigth I think ...

Reply:
@james7132 did you checked the lib impl for sqrtf [here](https://github.com/rust-lang/libm/blob/master/src/math/sqrtf.rs)? It don't look like it's cross-plataform deterministic

Reply:
I submitted a revision for this function, by the looks of it the compiler can optimize the re-alignment instructions so it should not be a problem

Reply:
It doesn't look like it at a first glance and it's difficult to tell if it actually produces the same output as the intrinsic with `sse` disabled. If it does, we may need to reevaluate this approach to achieving cross-platform FP determinism.

The `approx_rsqrt` implementation here, however, we for a fact know does not produce the same bit-for-bit results between the two implementations given the error difference seen from the benchmarks.

Reply:
I suggest you to resample the curve animations, that way you can mitigate any differences between cross plataform.